modelhub/modelhub/stack/util.py

```diff
@@ -1,53 +1,128 @@
 """
 Copyright 2021 Objectiv B.V.
 """
-from modelhub.stack.basic_features import BasicFeatures
-from modelhub.stack.extracted_contexts import ExtractedContexts
-from modelhub.stack.sessionized_data import SessionizedData
-
-from sql_models.model import SqlModel
-
-
-def _get_date_range(start_date, end_date):
-    if start_date:
-        if end_date:
-            date_range = f"where day between '{start_date}' and '{end_date}'"
-        else:
-            date_range = f"where day >= '{start_date}'"
-    else:
-        if end_date:
-            date_range = f"where day <= '{end_date}'"
-        else:
-            date_range = ''
-
-    return date_range
-
-
-def basic_feature_model(session_gap_seconds=1800,
-                        start_date=None,
-                        end_date=None,
-                        table_name='data') -> SqlModel:
-    """ Give a linked BasicFeatures model"""
-    date_range = _get_date_range(start_date, end_date)
-
-    extracted_contexts = ExtractedContexts(date_range=date_range, table_name=table_name)
-    return BasicFeatures.build(
-        sessionized_data=SessionizedData(
-            session_gap_seconds=session_gap_seconds,
-            extracted_contexts=extracted_contexts
-        )
+import bach
+
+from enum import Enum
+from typing import Tuple, Dict, List
+
+from modelhub.series import series_objectiv
+
+
+# Columns that Modelhub expects in an Objectiv dataframe
+class ObjectivSupportedColumns(Enum):
+    EVENT_ID = 'event_id'
+    DAY = 'day'
+    MOMENT = 'moment'
+    USER_ID = 'user_id'
+    GLOBAL_CONTEXTS = 'global_contexts'
+    LOCATION_STACK = 'location_stack'
+    EVENT_TYPE = 'event_type'
+    STACK_EVENT_TYPES = 'stack_event_types'
+    SESSION_ID = 'session_id'
+    SESSION_HIT_NUMBER = 'session_hit_number'
+
+    _DATA_SERIES = (
+        DAY, MOMENT, USER_ID, GLOBAL_CONTEXTS, LOCATION_STACK, EVENT_TYPE,
+        STACK_EVENT_TYPES, SESSION_ID, SESSION_HIT_NUMBER,
+    )
+
+    _INDEX_SERIES = (EVENT_ID, )
+
+    _EXTRACTED_CONTEXT_COLUMNS = (
+        EVENT_ID, DAY, MOMENT, USER_ID, GLOBAL_CONTEXTS, LOCATION_STACK, EVENT_TYPE, STACK_EVENT_TYPES,
+    )
+
+    _SESSIONIZED_COLUMNS = (
+        SESSION_ID, SESSION_HIT_NUMBER,
     )
 
+    @classmethod
+    def get_extracted_context_columns(cls) -> Tuple[str, ...]:
```

Comment:
I think I'll return lists instead, I keep casting the returns for all this functions
```suggestion
    def get_extracted_context_columns(cls) -> List[str]:
```